### DISCUSSION TITLE
Telegraf Connection Refused only from CrateDB Localhost

### DISCUSSION THREAD

**Discussion by blitzdakka** [crate/crate#12077] [Q&A]
Hi all,

I currently use Telegraf/InfluxDB/Grafana for my monitoring, but am looking at replacing InfluxDB with CrateDB. Currently I have all my machines feeding Telegraf metrics into CrateDB except for the VM running CrateDB itself, where Telegraf has the following error:

[agent] Failed to connect to [outputs.cratedb], retrying in 15s, error was 'failed to connect to `host=vm-cratedb user=crate database=doc`: dial error (dial tcp 127.0.1.1:5432: connect: connection refused)'

Every other machine with Telegraf is working correctly with the same configuration file.

Most of my machines, including the CrateDB host, are Debian 11 VMs running on Proxmox, however Telegraf is also working on the physical machine running Proxmox, as well as two Raspberry Pis. 

What I've tried:
- I am using the default 'crate' user, but have also tried setting up a new user with a password. 
- My network.host is set to _site_, I've tried _global_ also.
- Changing the connection string in Telegraf's conf to the DNS name, IP address, and loopback address.

Is there any configuration in CrateDB that might stop Telegraf connecting to the localhost? Any other troubleshooting I can try?

Many thanks.

**Comment by seut** [comment]
> My network.host is set to site, I've tried global also.
> 
> Is there any configuration in CrateDB that might stop Telegraf connecting to the localhost? Any other troubleshooting I can try?

`_site_` refers to any _site-local_ address -> any local network address inside 10/8, 172.16/12, 192.168/16 range
`_global_` refers to any _global_ address ->  any address NOT in the range above and NOT localhost

In order to listen to *any*  bound address you want to use `0.0.0.0`.
You can also set a list of values e.g. `_site_, 127.0.0.1`. 

See also https://crate.io/docs/crate/reference/en/4.6/config/node.html#hosts.

**Reply by blitzdakka** [reply]
Excellent thank you, that would explain it! I've read that docs page a few times, but unless I've gone blind, no-where does it specify that the setting can take multiple addresses.
